Main Cleanup of Workbench release notes (#32413)
mantidproject/mantid: docs/source/release/v6.2.0/mantidworkbench.rst

- Fixed a bug where parameters wouldn't update in the fit property browser when fitting a single function with ties.
- Fixed a bug where the workspace index spinbox in the fit browser wouldn't update when the user added or removed curves from the figure.
- Fixed a bug retrieving algorithm history from a workspace when the retrieval methods were chained together.
- Fixed a bug where output workspaces of different types would interfere with successive calls to binary operations, such as multiply.
- Fixed JSON serialization issue of MantidAxType by explicitly extracting its value.
- Fixed a bug where the errorbar tab in the figure options was wrongly enabled while selecting multiple curves.
- Fixed a bug where marker formatting options were disabled upon opening the figure options.
- Fixed the help icon not showing on OSX and high-resolution monitors.
- Tabbing between fields in the error reporter now works as expected, rather than jumping to a random place each time.
- Fixed a bug in the :ref:`InstrumentViewer` causing the projection to not be updated when different axis views were selected in Full 3D.

Sliceviewer
-----------

Improvements
############

> - Peaks can now be added or removed from a PeaksWorkspace using the :ref:`peaks overlay <sliceviewer_peaks_overlay>` in :ref:`sliceviewer`.

Reviewer: Thanks for grouping these, however, now it seems redundant to repeat :ref:`sliceviewer` in each item as they are all under one heading. Perhaps sublists would be better. The topmost item can say `Several improvements have been done in :ref:`sliceviewer`. Then the sub-items can just enumerate them. Peaks overlay can, of course, be referenced as it is a different link.

New and Improved

> ----------------

Reviewer: Could we split new from improved? There are 2 new widgets, it's better to make them stand out.

- **Superplot is a new decorator widget for the plot window. It facilitates over-plotting and manipulation of overplotted data. See** :ref:`Superplot documentation <WorkbenchSuperplot>` **for more information.**

.. figure:: ../../images/superplot_1.png
    :width: 500px
    :align: center

> - **New widget and workbench plugin:** :ref:`WorkbenchWorkspaceCalculator` **, allows to perform binary operations and scaling by a floating number on workspaces;

Reviewer 2: Missing a word here `New widget and workbench plugin: :ref:WorkbenchWorkspaceCalculator, allows (you/the user/users)`

> - **New widget and workbench plugin:** :ref:`WorkbenchWorkspaceCalculator` **, allows users to perform binary operations and scaling by a floating number on workspaces;

Reviewer: This sentence doesn't read well, could you rephrase it?

Author: I've had a look at the documentation for this which is similar to this release note but perhaps gives a little more detail - 'The workspace calculator allows to quickly perform binary operations on selected workspace pairs along with simultaneous scaling by arbitrary floating point numbers, provided they are different from zero.'

Maybe I should use this in the release notes instead?

Reviewer: Perhaps this is more concise:
`The workspace calculator allows to quickly perform binary operations on selected workspace pairs, optionally with scaling.`

  This will require your widget layout to be reset when starting workbench v6.2.0 for the first time. Previously saved layouts accessible from** ``View > User Layouts``
  **may need to be saved again to include the** :ref:`WorkbenchWorkspaceCalculator` **widget.**

.. image::  ../../images/Workbench/WorkspaceCalculator/WorkspaceCalculator.png
    :align: right

- Workflow diagrams in help pages are now ``.svg`` rather than ``.png``.
- Peaks can now be added or removed from a PeaksWorkspace using the :ref:`peaks overlay <sliceviewer_peaks_overlay>` in :ref:`sliceviewer`.
- When fitting a plot, selecting the peak type will only update the default peak shape in the settings if the ``Set as global default`` checkbox is ticked.
- Plot legends can be shown or hidden from the plot context menu.
- The plot config dialog notifies the user when there has been an error applying the config to the plot, and allows them to change the config further.

> - The list of eligible workspaces in the `WorkspaceSelector` can now be sorted by name.

Reviewer: This should be removed as the widget is new in this release.

Author: Is it? I thought the two new widgets were Superplot and the Workspace Calculator. Is the Workspace Selector the same as the Workspace Calculator? I confess I couldn't find any documentation on the Workspace Selector so wasn't sure what it was exactly.

Reviewer: ahh, sorry, you are right, the selector is as old as mantid

- Added tooltips to all the widgets in the :ref:`sliceviewer`. Please contact the developers if any are missing.

> - Script editor function call tip support for Numpy 1.21.

Reviewer 2: I feel like just saying code completion support covers everything, but maybe not.
```suggestion
- Script editor code completion support for Numpy 1.21.
```

- The visibility of a component parameter in the Pick tab of the :ref:`InstrumentViewer` is now steered by the 'visible' attribute of a parameter in :ref:`IPF <InstrumentParameterFile>`.

> - ADS signal handlers are now synchronized in `WorkspaceSelector`. This reduces the probability of hard crash when interacting with the widget while a script manipulating a large number of workspaces is being run.

Reviewer: This as well.

Reviewer: please ignore this as well

> - Added help button to the :ref:`sliceviewer`.

Reviewer 2: ```suggestion
- Added a help button to the :ref:`sliceviewer`.
```

> - :ref:`sliceviewer` can toggle between different scales again without any issue.

Reviewer 2: Should this be a bugfix?

- :ref:`sliceviewer` uses a more visible divider between the main data view and the peaks table view.

Bugfixes
--------

- ``Grid`` checkbox in ``Edit axis`` dialog, and ``Grids on/off`` toolbar button will now have the correct checked state when running a plot script with major grid lines.
- Fixed a bug in colorfill plots which lead to the loss of a spectrum from the resulting image.
- Fixed a bug where removing the plot guess line in the fit browser could lead to an exception being thrown.

> - Fixed the advanced plotting dialog incorrectly laying out, causing the options to be partially occluded.

Reviewer 2: ```suggestion
- Fixed the advanced plotting dialog incorrect layout, which caused options to be partially occluded.
```

- Fixed a bug where the workspace index spinbox in the fit browser wouldn't update when the user added or removed curves from the figure.
- Fixed a bug retrieving algorithm history from a workspace when the retrieval methods were chained together.
- Fixed a bug where output workspaces of different types would interfere with successive calls to binary operations, such as multiply.
- Fixed JSON serialization issue of MantidAxType by explicitly extracting its value.
- Fixed a bug in the :ref:`sliceviewer` when transposing MDE workspaces multiple times would cause the data to become all zeros.
- Fixed a bug where the errorbar tab in the figure options was wrongly enabled while selecting multiple curves.
- Fixed a bug where marker formatting options were disabled upon opening the figure options.
- Fixed out of range errors in the :ref:`sliceviewer` that sometimes occurred whilst hovering over transposed data.
- Fixed the help icon not showing on OSX and high-resolution monitors.
- Tabbing between fields in the error reporter now works as expected, rather than jumping to a random place each time.

> - :ref:`sliceviewer` can toggle between different scales again without any issue.

Reviewer: Can we group all the slice viewer improvements together?
